### PR TITLE
CAMEL-11476: spring-boot - routes not loaded when setting a management.port

### PR DESCRIPTION
**Pull request by zregvart**
@nicolaferraro would you mind taking a look, thanks 👍 

Fixes `RouteBuilder` (Java DSL) based routes not loading when setting `management.port` Spring Boot property by removing the 'same application context' check.

Also removes `maybeStart` method to start the Camel context from `RoutesCollector`, startup is now handled in `SpringCamelContext`.

When customizing the `management.port` configuration option Spring Boot creates a new `ApplicationContext` instance which when started emits a `ContextRefreshedEvent` that is different from the `ApplicationContext` the `RoutesCollector` is initialized with.

Due to change in CAMEL-11261, `SpringCamelContext` is started via the same `ContextRefreshedEvent`, at the point the second `ContextRefreshedEvent` emited from the `ApplicationContext`, that `RoutesCollector` is initialized with, `SpringCamelContext` is already started and no Java DSL routes are added to Camel context.

### CODE REVIEW

File: components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/RoutesCollector.java [hunk 1]
```diff
@@ -75,122 +75,111 @@ public RoutesCollector(ApplicationContext applicationContext, List<CamelContextC
...
                                 LOG.info("CamelSpringBoot will terminate after being idle for more {} seconds", configurationProperties.getDurationMaxIdleSeconds());
                             }
+                            // needed by MainDurationEventNotifier to signal when we have processed the max messages
+                            final AtomicBoolean completed = new AtomicBoolean();
+                            final CountDownLatch latch = new CountDownLatch(1);
+
                             // register lifecycle so we can trigger to shutdown the JVM when maximum number of messages has been processed
                             EventNotifier notifier = new MainDurationEventNotifier(camelContext,
                                 configurationProperties.getDurationMaxMessages(), configurationProperties.getDurationMaxIdleSeconds(),
-                                controller.getCompleted(), controller.getLatch(), true);
```
Comment (davsclaus) [inline components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/RoutesCollector.java]: Why dont you use like the old code, with those latches from the controller?

Comment (zregvart) [inline components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/RoutesCollector.java]: So I haven't changed anything but removed that `if (this.applicationContext.equals(applicationContext)) {` block, and removed `maybeStart` invocation and method. The diff is quite funky when you de-indent code, take a look at the [diff that ignores whitespace](https://github.com/apache/camel/pull/1797/files?w=1).

Comment (zregvart) [inline components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/RoutesCollector.java]: I could also split the PR into two commits one that introduces the change and one that formats the code, would that make it easier?

**Comment by zregvart** [comment]
Yeah, I've run `mvn clean install` and I've played with `camel-example-spring-boot` setting `management.port=8081` in `application.properties`.

Would like to hear from @nicolaferraro if removing that `if (this.applicationContext.equals(applicationContext)) {` would break anything else. Seems to me it was put there to prevent adding routes twice, with the new startup order in [CAMEL-11261](https://issues.apache.org/jira/browse/CAMEL-11261) the second `ContextRefreshedEvent` should find the `SpringCamelContext` started and skip adding any routes.

**Comment by nicolaferraro** [comment]
Looks good. Cannot try it at the moment but I think your debugging session has been long enough :)
I'd merge it if tests pass and I'll test it on the failing quickstarts...

**Pull request closed without merging**